### PR TITLE
Segregate all the latest XTZ fixes

### PR DESCRIPTION
**Pull request by alekece**
This PR embeds all the latest XTZ fixes in an exhaustive way.

 **Disclaimer**: Maybe some stuff are missing but still we don't have XTZ tests, we cannot properly check if all are here.

## Mandatory picture of a dog
mechano_dog

### CODE REVIEW

File: ledger-core-tezos/src/tezos/transactions/TezosLikeTransaction.cpp [hunk 1]
```diff
@@ -153,6 +171,19 @@ namespace ledger {
                 writer.writeByte(static_cast<uint8_t>(api::TezosOperationTag::OPERATION_TAG_GENERIC));
             }
 
+            // If tx was forged then nothing to do
+            if (!_rawTx.empty()) {
+                writer.writeByteArray(_rawTx);
```
Comment (KhalilBellakrid) [inline ledger-core-tezos/src/tezos/transactions/TezosLikeTransaction.cpp]: I think you are missing a fix here : https://github.com/LedgerHQ/lib-ledger-core/commit/672428c5920047d0d20f3fe418f77b2ace46b110

File: ledger-core-tezos/src/tezos/transactions/TezosLikeTransaction.cpp [hunk 2]
```diff
@@ -162,25 +193,54 @@ namespace ledger {
...
             // Set Sender
-            // Originated
-            auto isSenderOriginated = _sender->toBase58().find("KT1") == 0;
-            writer.writeByte(static_cast<uint8_t>(isSenderOriginated));
-            auto senderContractID = isSenderOriginated ?
-                                    vector::concat(_sender->getHash160(), {0x00}) :
-                                    vector::concat({static_cast<uint8_t>(_senderCurve)}, _sender->getHash160());
-            writer.writeByteArray(senderContractID);
+            if (isBabylonActivated) {
+                auto senderContractID = vector::concat({static_cast<uint8_t>(_senderCurve)}, _sender->getHash160());
```
Comment (KhalilBellakrid) [inline ledger-core-tezos/src/tezos/transactions/TezosLikeTransaction.cpp]: Same here : https://github.com/LedgerHQ/lib-ledger-core/commit/50f646f69e291be570b14bb562e98566444e4928

File: ledger-core-tezos/src/tezos/transactions/TezosLikeTransaction.cpp [hunk 3]
```diff
@@ -189,16 +249,15 @@ namespace ledger {
             // Storage Limit
             writer.writeByteArray(zarith::zSerializeNumber(_storage->toByteArray()));
```
Comment (KhalilBellakrid) [inline ledger-core-tezos/src/tezos/transactions/TezosLikeTransaction.cpp]: Same : https://github.com/LedgerHQ/lib-ledger-core/commit/e3e93ffd023d53a1b2ef0bb33a5fb425332a41f1#diff-470f4114b9e3d3f052e9177c968e5959R260

**Comment by alekece** [comment]
Thanks you for your wise review. I took into consideration your comments - all the points you got back are already pushed - then I will read back the `TezosLikeAccount2` to be sure is match the current one on `develop`

**Comment by phaazon** [comment]
Thank you so much @alekece, very nice work! I’ll merge this, as we will anyway review the process `design/modularization -> develop`.

**Comment by phaazon** [comment]
Also, if you have other things to merge from Tezos, feel free to open another PR. I stand by the idea that PRs should be minimal and easy to read.

**Pull request merged**